pxtsim/simdriver.ts

```diff
@@ -77,12 +77,22 @@ namespace pxsim {
         public state = SimulatorState.Unloaded;
         public hwdbg: HwDebugger;
         private _dependentEditors: Window[];
+        private _allowedOrigins: Set<string> = new Set();
 
         // we might "loan" a simulator when the user is recording
         // screenshots for sharing
         private loanedSimulator: HTMLDivElement;
 
         constructor(public container: HTMLElement, public options: SimulatorDriverOptions = {}) {
+            this._allowedOrigins.add(window.location.origin);
+            if (options.parentOrigin)
+            this._allowedOrigins.add(options.parentOrigin)
+            try {
+                const simUrl = new URL(this.getSimUrl())
+                this._allowedOrigins.add(simUrl.origin)
+            } catch (e) {
+                console.error(`Invalid sim url ${this.getSimUrl()}`)
```

Comment:
use ``pxt.reportException``

Reply:
I've made the change, but I had to include `/// <reference path="../built/pxtlib.d.ts"/>` if that is something to consider.

Reply:
Okay, I was told that I should not include `pxtlib` in `pxtsim` due to how much that will increase the file size of `pxtsim`. Without referencing the lib, the `reportException` function is not available to me. I'll reverse my change to just use the console for the time being.

It seems there is work to introduce analytics to the sim in a more lightweight way than including all of pxtlib. Thoughts @pelikhan?